src/encoder.rs

```diff
@@ -2066,17 +2150,50 @@ fn encode_tile_group<T: Pixel>(fi: &FrameInvariants<T>, fs: &mut FrameState<T>)
     fs.t.print_code();
   }
 
-  fs.cdfs = fc;
+  let (idx_max, max_len) = raw_tiles
+    .iter()
+    .map(Vec::len)
+    .enumerate()
+    .max_by_key(|&(_, len)| len)
+    .unwrap();
+
+  // use the biggest tile (in bytes) for CDF update
+  fs.context_update_tile_id = idx_max;
+  fs.cdfs = cdfs[idx_max];
   fs.cdfs.reset_counts();
 
-  data
+  let max_tile_size_bytes = ((max_len as u32).ilog() + 7) / 8;
+  debug_assert!(max_tile_size_bytes > 0 && max_tile_size_bytes <= 4);
+  fs.max_tile_size_bytes = max_tile_size_bytes;
```

Comment:
any reason this is stored in framestate?

Reply:
This is a property at frame level (there is 1 `FrameState` per frame, and 1 `TileState` per tile, i.e. _n_ `TileState`s per frame).

It is then retrieved from `FrameState` to write the header (see commit `Optimizer tile size bytes in bitstream`, currently 1558b052119eaf58fdac48eec661d2f28be5918b).

Reply:
OK. Makes me somewhat uncomfortable how it sits invalid for so long, but I guess the assert for bytes > 0 should cover that case.

Reply:
> Makes me somewhat uncomfortable how it sits invalid for so long

This is exactly the same for the `cdfs` variable in `FrameState`, it is initialized only there (4 lines above, `fs.cdfs = cdfs;`).